File: binary_search/69. Sqrt(x)/69.js
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    if (x === 1) return x
        let l = 0
        let r = x
        while ( (r - l) > 1){
            m = Math.floor((r + l) / 2)
            if (m ** 2 == x) return m
            if (m ** 2 > x) r = m
            if (m ** 2 < x) l = m
        }

        return l
};

File: binary_search/69. Sqrt(x)/69.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if (x == 1): return x
        l = 0
        r = x
        while ( (r - l) > 1):
            m = (r + l) // 2
            if (m ** 2 == x): return m
            if m ** 2 > x: r = m
            if (m ** 2 < x): l = m

        return l